crates/bevy_asset/src/processor/mod.rs

```diff
@@ -0,0 +1,1265 @@
+                    started_processing = true;
+                }
+
+                self.handle_asset_source_event(event).await;
+            }
+
+            if started_processing {
+                self.finish_processing_assets().await;
+            }
+        }
+    }
+
+    async fn handle_asset_source_event(&self, event: AssetSourceEvent) {
+        match event {
+            AssetSourceEvent::AddedAsset(path)
+            | AssetSourceEvent::AddedMeta(path)
+            | AssetSourceEvent::ModifiedAsset(path)
+            | AssetSourceEvent::ModifiedMeta(path) => {
+                self.process_asset(&path).await;
+            }
+            AssetSourceEvent::RemovedAsset(path) => {
+                self.handle_removed_asset(path).await;
+            }
+            AssetSourceEvent::RemovedMeta(path) => {
+                self.handle_removed_meta(&path).await;
+            }
+            AssetSourceEvent::AddedFolder(path) => {
+                self.handle_added_folder(path).await;
+            }
+            // NOTE: As a heads up for future devs: this event shouldn't be run in parallel with other events that might
+            // touch this folder (ex: the folder might be re-created with new assets). Clean up the old state first.
+            // Currently this event handler is not parallel, but it could be (and likely should be) in the future.
+            AssetSourceEvent::RemovedFolder(path) => {
+                self.handle_removed_folder(&path).await;
+            }
+            AssetSourceEvent::RenamedAsset { old, new } => {
+                // If there was a rename event, but the path hasn't changed, this asset might need reprocessing.
+                // Sometimes this event is returned when an asset is moved "back" into the asset folder
+                if old == new {
+                    self.process_asset(&new).await;
+                } else {
+                    self.handle_renamed_asset(old, new).await;
+                }
+            }
+            AssetSourceEvent::RenamedMeta { old, new } => {
+                // If there was a rename event, but the path hasn't changed, this asset meta might need reprocessing.
+                // Sometimes this event is returned when an asset meta is moved "back" into the asset folder
+                if old == new {
+                    self.process_asset(&new).await;
+                } else {
+                    debug!("Meta renamed from {old:?} to {new:?}");
+                    let mut infos = self.data.asset_infos.write().await;
+                    // Renaming meta should not assume that an asset has also been renamed. Check both old and new assets to see
+                    // if they should be re-imported (and/or have new meta generated)
+                    infos.check_reprocess_queue.push_back(old);
+                    infos.check_reprocess_queue.push_back(new);
+                }
+            }
+            AssetSourceEvent::RenamedFolder { old, new } => {
```

Comment:
Hmmm I'll investigate. Possible I missed a case or mishandled it. I've spent much more time on the linux side.

Reply:
Cool I repro-ed! I'm pretty sure this is a "cleanup" step failing (because it was already cleaned up). I don't think anything is "broken" per-se. The fix will probably either be to not do the cleanup step in this case or to suppress the error.

Reply:
Yup! This was happening because when we remove the last individual asset from a destination folder, leaving the folder empty, we automatically clean up the folder. So this final folder remove is redundant in that case. Its worth keeping the duplicate remove in the event of a fully empty folder, so I just suppressed the error in the NotFound case.